### PR TITLE
Implemented basic backend+frontend flow of Visualizer.

### PR DESCRIPTION
**Pull request by DSep**
Added backend to get course and related courses. Added visualizer front end to get and display courses. Other small edits.

Co-authored-by: Brandon @beetai

### CODE REVIEW

File: frontend/src/components/VisualCourse/courses.tsx [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import Course from './course';
-
+import Course from './Course';
 
+// REVIEW and delete if not necessary. Unused/WIP Components don't need to be committed.
```
Comment (baconandchips) [inline frontend/src/components/VisualCourse/courses.tsx]: Possible candidate for deleting, yes; I think it's un-used right now

**Review by baconandchips — APPROVED** [review]
LGTM, you can delete components/VisualCourse/courses.tsx if you wish to

**Review by yqj2k — APPROVED** [review]
LGTM, I remember seeing a small demo on this yesterday after class.

**Comment by baconandchips** [comment]
LGTM

**Pull request merged**